internal/model/manifest.go

```diff
@@ -214,7 +215,12 @@ func (m Manifest) Dependencies() []string {
 		deps = append(deps, f)
 	}
 
-	return deps
+	return sliceutils.DedupeStringSlice(deps)
+}
+
+func (m Manifest) WithDependencies(deps []string) Manifest {
```

Comment:
hmm, I think this should be `WithConfigFiles`. Dependencies aren't an explicit field on the struct, they're calculated from fields. While on globalyaml these are one-to-one, on regular manifests they are composed of many fields.

Just a nit, though.

Reply:
Nah, good call, will change